odbc/src/api/functions.rs

```diff
@@ -1215,12 +1340,210 @@ pub extern "C" fn SQLSetStmtAttr(
+                SqlReturn::ERROR
+            }
+            StatementAttribute::ImpParamDesc => {
+                stmt_handle.add_diag_info(ODBCError::Unimplemented("SQL_ATTR_IMP_PARAM_DESC"));
+                SqlReturn::ERROR
+            }
+            StatementAttribute::CursorScrollable => {
+                match FromPrimitive::from_usize(value as usize) {
+                    Some(CursorScrollable::NonScrollable) => SqlReturn::SUCCESS,
+                    _ => {
+                        stmt_handle.add_diag_info(ODBCError::InvalidAttrValue(
+                            "SQL_ATTR_CURSOR_SCROLLABLE",
+                        ));
+                        SqlReturn::ERROR
+                    }
+                }
+            }
+            StatementAttribute::CursorSensitivity => match FromPrimitive::from_i32(value as i32) {
+                Some(CursorSensitivity::Insensitive) => SqlReturn::SUCCESS,
+                _ => {
+                    stmt_handle
+                        .add_diag_info(ODBCError::InvalidAttrValue("SQL_ATTR_CURSOR_SENSITIVITY"));
+                    SqlReturn::ERROR
+                }
+            },
+            StatementAttribute::AsyncEnable => {
+                stmt_handle.add_diag_info(ODBCError::Unimplemented("SQL_ATTR_ASYNC_ENABLE"));
+                SqlReturn::ERROR
+            }
+            StatementAttribute::Concurrency => match FromPrimitive::from_i32(value as i32) {
+                Some(Concurrency::ReadOnly) => SqlReturn::SUCCESS,
+                _ => {
+                    stmt_handle.add_diag_info(ODBCError::OptionValueChanged(
+                        "SQL_ATTR_CONCURRENCY",
+                        "SQL_CONCUR_READ_ONLY",
+                    ));
+                    SqlReturn::SUCCESS_WITH_INFO
+                }
+            },
+            StatementAttribute::CursorType => match FromPrimitive::from_i32(value as i32) {
+                Some(CursorType::ForwardOnly) => SqlReturn::SUCCESS,
+                _ => {
+                    stmt_handle.add_diag_info(ODBCError::OptionValueChanged(
+                        "SQL_ATTR_CURSOR_TYPE",
+                        "SQL_CURSOR_FORWARD_ONLY",
+                    ));
+                    SqlReturn::SUCCESS_WITH_INFO
+                }
+            },
+            StatementAttribute::EnableAutoIpd => {
+                stmt_handle.add_diag_info(ODBCError::Unimplemented("SQL_ATTR_ENABLE_AUTO_IPD"));
+                SqlReturn::ERROR
+            }
+            StatementAttribute::FetchBookmarkPtr => {
+                stmt_handle.add_diag_info(ODBCError::Unimplemented("SQL_ATTR_FETCH_BOOKMARK_PTR"));
+                SqlReturn::ERROR
+            }
+            StatementAttribute::KeysetSize => {
+                stmt_handle.add_diag_info(ODBCError::Unimplemented("SQL_ATTR_KEYSET_SIZE"));
+                SqlReturn::ERROR
+            }
+            StatementAttribute::MaxLength => {
```

Comment:
Oh hmm, perhaps not. If an application wants to truncate data it can "specify the maximum buffer length in the BufferLength argument in SQLBindCol or SQLGetData," which seems sufficient.

Reply:
Yes, I also thinks this is what we need, because I don't think we can limit the network traffic by passing this to the Rust driver and ask to only stream that much data.
Sounds good to me :)